app/models/user.rb

```diff
@@ -2,11 +2,12 @@ class User < ApplicationRecord
   include Resources
   include WhitespaceValidation
 
+  has_secure_password
   has_many :tickets, dependent: :nullify
 
   USER_ROLE = { customer: 0, admin: 1 }.freeze
   enum role: USER_ROLE
-  validates :email, :user_name, :password, :phone, :role, presence: true
+  validates :email, :user_name, :password_digest, :phone, :role, presence: true
```

Comment:
is it mandatory to validate password_digest field?

Reply:
Thanks for the feedback bhaia. `password` should be validated instead of `password_digest`. Updated it.